Meson incorrectly installing like a darwin in an archlinux makepkg environment

**Describe the bug**

When `ninja install`-ing sway package in a fakeroot (as run by `makepkg`) on Archlinux, meson is taking some path that should be reserved to darwin systems and try to call `otool`. I do have cross-compilation tools installed on my system for the darwin platform. Assigned to meson since `sway` project doesn't have reference to `otool` in its repository.

Full logs and stacktrace below:

```
~/workspace/archofficiel/community/sway/repos/community-x86_64 ‹master› 
╰─$ makepkg                                                                                                                         1 ↵
==> Making package: sway 1:1.5.1-1 (Wed 25 Nov 2020 12:09:10 PM CET)
==> Checking runtime dependencies...
==> Checking buildtime dependencies...
==> Retrieving sources...
  -> Found sway-1.5.1.tar.gz
  -> Found sway-1.5.1.tar.gz.sig
  -> Found 50-systemd-user.conf
==> Validating source files with sha512sums...
    sway-1.5.1.tar.gz ... Passed
    sway-1.5.1.tar.gz.sig ... Skipped
    50-systemd-user.conf ... Passed
==> Verifying source file signatures with gpg...
    sway-1.5.1.tar.gz ... Passed
==> Extracting sources...
  -> Extracting sway-1.5.1.tar.gz with bsdtar
==> Starting prepare()...
==> Removing existing $pkgdir/ directory...
==> Starting build()...
+ exec meson setup --prefix /usr --libexecdir lib --sbindir bin --buildtype plain --auto-features enabled --wrap-mode nodownload -D b_lto=true -D b_pie=true build sway-1.5.1 -D werror=false -D b_ndebug=true
Directory already configured.

Just run your build command (e.g. ninja) and Meson will regenerate as necessary.
If ninja fails, run "ninja reconfigure" or "meson --reconfigure"
to force Meson to regenerate.

If build failures persist, run "meson setup --wipe" to rebuild from scratch
using the same options as passed when configuring the build.
To change option values, run "meson configure" instead.
ninja: Entering directory `build'
[0/1] Regenerating build files.
The Meson build system
Version: 0.56.0
Source dir: /home/alexandre/workspace/archofficiel/community/sway/repos/community-x86_64/src/sway-1.5.1
Build dir: /home/alexandre/workspace/archofficiel/community/sway/repos/community-x86_64/src/build
Build type: native build
Project name: sway
Project version: 1.5.1
C compiler for the host machine: ccache gcc (gcc 10.2.0 "gcc (GCC) 10.2.0")
C linker for the host machine: gcc ld.bfd 2.35.1
Host machine cpu family: x86_64
Host machine cpu: x86_64
Dependency json-c found: YES 0.15 (cached)
Dependency libpcre found: YES 8.44 (cached)
Dependency wayland-server found: YES 1.18.0 (cached)
Dependency wayland-client found: YES 1.18.0 (cached)
Dependency wayland-cursor found: YES 1.18.0 (cached)
Dependency wayland-egl found: YES 18.1.0 (cached)
Dependency wayland-protocols found: YES 1.20 (cached)
Dependency xkbcommon found: YES 1.0.2 (cached)
Dependency cairo found: YES 1.17.3 (cached)
Dependency pango found: YES 1.48.0 (cached)
Dependency pangocairo found: YES 1.48.0 (cached)
Dependency gdk-pixbuf-2.0 found: YES 2.42.0 (cached)
Dependency pixman-1 found: YES 0.40.0 (cached)
Dependency glesv2 found: YES 3.2 (cached)
Dependency libevdev found: YES 1.10.0 (cached)
Dependency libinput found: YES 1.16.3 (cached)
Dependency libsystemd found: YES 246 (cached)
Found pkg-config: /usr/bin/pkg-config (1.7.3)
Found CMake: /usr/bin/cmake (3.19.0)
Run-time dependency libelogind found: NO (tried pkgconfig and cmake)
Dependency xcb found: YES 1.14 (cached)
Run-time dependency bash-completion found: NO (tried pkgconfig and cmake)
Run-time dependency fish found: NO (tried pkgconfig and cmake)
Library m found: YES
Library rt found: YES
Neither a subproject directory nor a wlroots.wrap file was found.
Subproject  wlroots is buildable: NO (disabling)
Dependency wlroots found: YES 0.12.0 (cached)
Fetching value of define "WLR_HAS_XWAYLAND" with dependency wlroots: 1 (cached)
Dependency scdoc found: YES 1.11.0 (cached)
Program /usr/bin/scdoc found: YES (/usr/bin/scdoc)
Program sh found: YES (/usr/bin/sh)
Program git found: YES (/usr/bin/git)
Compiler for C supports arguments -fmacro-prefix-map=/prefix/to/hide=: YES (cached)
Configuring config.h using configuration
Dependency wayland-scanner found: YES 1.18.0 (cached)
Program /usr/bin/wayland-scanner found: YES (/usr/bin/wayland-scanner)
Configuring config using configuration
Build targets in project: 38

sway 1.5.1

      xwayland: YES
    gdk-pixbuf: YES
       systemd: YES
       elogind: NO
          tray: YES
     man-pages: YES

  Subprojects
       wlroots: NO Neither a subproject directory nor a wlroots.wrap file was found.

Option werror is: false [default: true]
Found ninja-1.10.1 at /usr/bin/ninja
ninja: no work to do.
==> Entering fakeroot environment...
==> Starting package()...
ninja: Entering directory `build'
[0/1] Installing files.
Installing sway.1 to /home/alexandre/workspace/archofficiel/community/sway/repos/community-x86_64/pkg/sway/usr/share/man/man1
Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/mesonbuild/mesonmain.py", line 140, in run
    return options.run_func(options)
  File "/usr/lib/python3.8/site-packages/mesonbuild/minstall.py", line 554, in run
    installer.do_install(datafilename)
  File "/usr/lib/python3.8/site-packages/mesonbuild/minstall.py", line 372, in do_install
    self.install_targets(d)
  File "/usr/lib/python3.8/site-packages/mesonbuild/minstall.py", line 527, in install_targets
    depfixer.fix_rpath(outname, t.rpath_dirs_to_remove, install_rpath, final_path,
  File "/usr/lib/python3.8/site-packages/mesonbuild/scripts/depfixer.py", line 484, in fix_rpath
    fix_darwin(fname, new_rpath, final_path, install_name_mappings)
  File "/usr/lib/python3.8/site-packages/mesonbuild/scripts/depfixer.py", line 396, in fix_darwin
    rpaths = get_darwin_rpaths_to_remove(fname)
  File "/usr/lib/python3.8/site-packages/mesonbuild/scripts/depfixer.py", line 377, in get_darwin_rpaths_to_remove
    out = subprocess.check_output(['otool', '-l', fname],
  File "/usr/lib/python3.8/subprocess.py", line 411, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
  File "/usr/lib/python3.8/subprocess.py", line 489, in run
    with Popen(*popenargs, **kwargs) as process:
  File "/usr/lib/python3.8/subprocess.py", line 854, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/usr/lib/python3.8/subprocess.py", line 1702, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'otool'
FAILED: meson-install 
/usr/bin/meson install --no-rebuild
ninja: build stopped: subcommand failed.
==> ERROR: A failure occurred in package().
    Aborting...
```

**To Reproduce**

With makepkg tools installed (like, on archlinux):

```
git clone https://github.com/archlinux/svntogit-community community
cd community/sway/repos/community-x86_64/
makepkg
```
I didn't try on a clean archlinux install yet.

**Expected behavior**

The package install normally.

**system parameters**
* This is a plain native build
* Up-to-date archlinux
* Python version  3.8.6
* Archlinux upstream meson version 0.56.0
* Ninja version 1.10,1

Full `meson-logs.txt` (from a clean `makepkg`):

```
Build started at 2020-11-25T12:44:18.342654
Main binary: /usr/bin/python
Build Options: -Db_lto=true -Db_pie=true -Dwerror=false -Db_ndebug=true -Dprefix=/usr -Dlibexecdir=lib -Dsbindir=bin -Dauto_features=enabled -Dbuildtype=plain -Dwrap_mode=nodownload
Python system: Linux
The Meson build system
Version: 0.56.0
Source dir: /home/alexandre/workspace/archofficiel/community/sway/repos/community-x86_64/src/sway-1.5.1
Build dir: /home/alexandre/workspace/archofficiel/community/sway/repos/community-x86_64/src/build
Build type: native build
Project name: sway
Project version: 1.5.1
Using 'CC' from environment with value: 'ccache gcc'
Using 'CFLAGS' from environment with value: '-march=x86-64 -mtune=generic -O2 -pipe -fno-plt -g -fvar-tracking-assignments -fdebug-prefix-map=/home/alexandre/workspace/archofficiel/community/sway/repos/community-x86_64/src=/usr/src/debug'
Using 'LDFLAGS' from environment with value: '-Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now'
Using 'CPPFLAGS' from environment with value: '-D_FORTIFY_SOURCE=2'
None of 'CC_LD' are defined in the environment, not changing global flags.
Sanity testing C compiler: ccache gcc
Is cross compiler: False.
None of 'CC_LD' are defined in the environment, not changing global flags.
Sanity check compiler command line: ccache gcc /home/alexandre/workspace/archofficiel/community/sway/repos/community-x86_64/src/build/meson-private/sanitycheckc.c -o /home/alexandre/workspace/archofficiel/community/sway/repos/community-x86_64/src/build/meson-private/sanitycheckc.exe -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -g -fvar-tracking-assignments -fdebug-prefix-map=/home/alexandre/workspace/archofficiel/community/sway/repos/community-x86_64/src=/usr/src/debug -D_FORTIFY_SOURCE=2 -pipe -D_FILE_OFFSET_BITS=64 -Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now
Sanity check compile stdout:

-----
Sanity check compile stderr:

-----
Running test binary command: /home/alexandre/workspace/archofficiel/community/sway/repos/community-x86_64/src/build/meson-private/sanitycheckc.exe
C compiler for the host machine: ccache gcc (gcc 10.2.0 "gcc (GCC) 10.2.0")
C linker for the host machine: gcc ld.bfd 2.35.1
None of 'AR' are defined in the environment, not changing global flags.
Using 'CC' from environment with value: 'ccache gcc'
Using 'CFLAGS' from environment with value: '-march=x86-64 -mtune=generic -O2 -pipe -fno-plt -g -fvar-tracking-assignments -fdebug-prefix-map=/home/alexandre/workspace/archofficiel/community/sway/repos/community-x86_64/src=/usr/src/debug'
Using 'LDFLAGS' from environment with value: '-Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now'
Using 'CPPFLAGS' from environment with value: '-D_FORTIFY_SOURCE=2'
None of 'CC_LD' are defined in the environment, not changing global flags.
Sanity testing C compiler: ccache gcc
Is cross compiler: False.
None of 'CC_LD' are defined in the environment, not changing global flags.
Sanity check compiler command line: ccache gcc /home/alexandre/workspace/archofficiel/community/sway/repos/community-x86_64/src/build/meson-private/sanitycheckc.c -o /home/alexandre/workspace/archofficiel/community/sway/repos/community-x86_64/src/build/meson-private/sanitycheckc.exe -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -g -fvar-tracking-assignments -fdebug-prefix-map=/home/alexandre/workspace/archofficiel/community/sway/repos/community-x86_64/src=/usr/src/debug -D_FORTIFY_SOURCE=2 -pipe -D_FILE_OFFSET_BITS=64 -Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now
Sanity check compile stdout:

-----
Sanity check compile stderr:

-----
Running test binary command: /home/alexandre/workspace/archofficiel/community/sway/repos/community-x86_64/src/build/meson-private/sanitycheckc.exe
C compiler for the build machine: ccache gcc (gcc 10.2.0 "gcc (GCC) 10.2.0")
C linker for the build machine: gcc ld.bfd 2.35.1
None of 'AR' are defined in the environment, not changing global flags.
Build machine cpu family: x86_64
Build machine cpu: x86_64
Host machine cpu family: x86_64
Host machine cpu: x86_64
Target machine cpu family: x86_64
Target machine cpu: x86_64
Pkg-config binary for MachineChoice.HOST is not cached.
None of 'PKG_CONFIG' are defined in the environment, not changing global flags.
Pkg-config binary missing from cross or native file, or env var undefined.
Trying a default Pkg-config fallback at pkg-config
Found pkg-config: /usr/bin/pkg-config (1.7.3)
Determining dependency 'json-c' with pkg-config executable '/usr/bin/pkg-config'
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config --modversion json-c` -> 0
0.15
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config --cflags json-c` -> 0
-I/usr/include/json-c
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config json-c --libs` -> 0
-L/usr/lib -ljson-c
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config json-c --libs` -> 0
-ljson-c
None of 'PKG_CONFIG_PATH' are defined in the environment, not changing global flags.
None of 'CC_LD' are defined in the environment, not changing global flags.
Running compile:
Working directory:  /home/alexandre/workspace/archofficiel/community/sway/repos/community-x86_64/src/build/meson-private/tmpf_c6pi1g
Command line:  ccache gcc /home/alexandre/workspace/archofficiel/community/sway/repos/community-x86_64/src/build/meson-private/tmpf_c6pi1g/testfile.c -o /home/alexandre/workspace/archofficiel/community/sway/repos/community-x86_64/src/build/meson-private/tmpf_c6pi1g/output.exe -pipe -march=x86-64 -mtune=generic -O2 -fno-plt -g -fvar-tracking-assignments -fdebug-prefix-map=/home/alexandre/workspace/archofficiel/community/sway/repos/community-x86_64/src=/usr/src/debug -D_FORTIFY_SOURCE=2 -D_FILE_OFFSET_BITS=64 -O0 -Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now 

Code:
 #include<stdio.h>
        
        int main(void) {
            printf("%ld\n", (long)(sizeof(void *)));
            return 0;
        };
Compiler stdout:
 
Compiler stderr:
 In file included from /usr/include/bits/libc-header-start.h:33,
                 from /usr/include/stdio.h:27,
                 from /home/alexandre/workspace/archofficiel/community/sway/repos/community-x86_64/src/build/meson-private/tmpf_c6pi1g/testfile.c:1:
/usr/include/features.h:397:4: warning: #warning _FORTIFY_SOURCE requires compiling with optimization (-O) [-Wcpp]
  397 | #  warning _FORTIFY_SOURCE requires compiling with optimization (-O)
      |    ^~~~~~~

Program stdout:

8

Program stderr:


Running compile:
Working directory:  /home/alexandre/workspace/archofficiel/community/sway/repos/community-x86_64/src/build/meson-private/tmpq4dsorzy
Command line:  ccache gcc /home/alexandre/workspace/archofficiel/community/sway/repos/community-x86_64/src/build/meson-private/tmpq4dsorzy/testfile.c -o /home/alexandre/workspace/archofficiel/community/sway/repos/community-x86_64/src/build/meson-private/tmpq4dsorzy/output.obj -pipe -c -march=x86-64 -mtune=generic -O2 -fno-plt -g -fvar-tracking-assignments -fdebug-prefix-map=/home/alexandre/workspace/archofficiel/community/sway/repos/community-x86_64/src=/usr/src/debug -D_FORTIFY_SOURCE=2 -D_FILE_OFFSET_BITS=64 -O0 --print-search-dirs 

Code:
 
Compiler stdout:
 install: /usr/lib/gcc/x86_64-pc-linux-gnu/10.2.0/
programs: =/usr/lib/gcc/x86_64-pc-linux-gnu/10.2.0/:/usr/lib/gcc/x86_64-pc-linux-gnu/10.2.0/:/usr/lib/gcc/x86_64-pc-linux-gnu/:/usr/lib/gcc/x86_64-pc-linux-gnu/10.2.0/:/usr/lib/gcc/x86_64-pc-linux-gnu/:/usr/lib/gcc/x86_64-pc-linux-gnu/10.2.0/../../../../x86_64-pc-linux-gnu/bin/x86_64-pc-linux-gnu/10.2.0/:/usr/lib/gcc/x86_64-pc-linux-gnu/10.2.0/../../../../x86_64-pc-linux-gnu/bin/
libraries: =/usr/lib/gcc/x86_64-pc-linux-gnu/10.2.0/:/usr/lib/gcc/x86_64-pc-linux-gnu/10.2.0/../../../../x86_64-pc-linux-gnu/lib/x86_64-pc-linux-gnu/10.2.0/:/usr/lib/gcc/x86_64-pc-linux-gnu/10.2.0/../../../../x86_64-pc-linux-gnu/lib/../lib/:/usr/lib/gcc/x86_64-pc-linux-gnu/10.2.0/../../../x86_64-pc-linux-gnu/10.2.0/:/usr/lib/gcc/x86_64-pc-linux-gnu/10.2.0/../../../../lib/:/lib/x86_64-pc-linux-gnu/10.2.0/:/lib/../lib/:/usr/lib/x86_64-pc-linux-gnu/10.2.0/:/usr/lib/../lib/:/usr/lib/gcc/x86_64-pc-linux-gnu/10.2.0/../../../../x86_64-pc-linux-gnu/lib/:/usr/lib/gcc/x86_64-pc-linux-gnu/10.2.0/../../../:/lib/:/usr/lib/

Compiler stderr:
 
Run-time dependency json-c found: YES 0.15
Pkg-config binary for MachineChoice.HOST is cached.
Determining dependency 'libpcre' with pkg-config executable '/usr/bin/pkg-config'
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config --modversion libpcre` -> 0
8.44
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config --cflags libpcre` -> 0

PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config libpcre --libs` -> 0
-L/usr/lib -lpcre
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config libpcre --libs` -> 0
-lpcre
None of 'PKG_CONFIG_PATH' are defined in the environment, not changing global flags.
Run-time dependency libpcre found: YES 8.44
Pkg-config binary for MachineChoice.HOST is cached.
Determining dependency 'wayland-server' with pkg-config executable '/usr/bin/pkg-config'
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config --modversion wayland-server` -> 0
1.18.0
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config --cflags wayland-server` -> 0

PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config wayland-server --libs` -> 0
-L/usr/lib -lwayland-server
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config wayland-server --libs` -> 0
-lwayland-server
None of 'PKG_CONFIG_PATH' are defined in the environment, not changing global flags.
Run-time dependency wayland-server found: YES 1.18.0
Pkg-config binary for MachineChoice.HOST is cached.
Determining dependency 'wayland-client' with pkg-config executable '/usr/bin/pkg-config'
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config --modversion wayland-client` -> 0
1.18.0
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config --cflags wayland-client` -> 0

PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config wayland-client --libs` -> 0
-L/usr/lib -lwayland-client
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config wayland-client --libs` -> 0
-lwayland-client
None of 'PKG_CONFIG_PATH' are defined in the environment, not changing global flags.
Run-time dependency wayland-client found: YES 1.18.0
Pkg-config binary for MachineChoice.HOST is cached.
Determining dependency 'wayland-cursor' with pkg-config executable '/usr/bin/pkg-config'
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config --modversion wayland-cursor` -> 0
1.18.0
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config --cflags wayland-cursor` -> 0

PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config wayland-cursor --libs` -> 0
-L/usr/lib -lwayland-cursor
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config wayland-cursor --libs` -> 0
-lwayland-cursor
None of 'PKG_CONFIG_PATH' are defined in the environment, not changing global flags.
Run-time dependency wayland-cursor found: YES 1.18.0
Pkg-config binary for MachineChoice.HOST is cached.
Determining dependency 'wayland-egl' with pkg-config executable '/usr/bin/pkg-config'
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config --modversion wayland-egl` -> 0
18.1.0
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config --cflags wayland-egl` -> 0

PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config wayland-egl --libs` -> 0
-L/usr/lib -lwayland-egl -lwayland-client
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config wayland-egl --libs` -> 0
-lwayland-egl -lwayland-client
None of 'PKG_CONFIG_PATH' are defined in the environment, not changing global flags.
Run-time dependency wayland-egl found: YES 18.1.0
Pkg-config binary for MachineChoice.HOST is cached.
Determining dependency 'wayland-protocols' with pkg-config executable '/usr/bin/pkg-config'
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config --modversion wayland-protocols` -> 0
1.20
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config --cflags wayland-protocols` -> 0

PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config wayland-protocols --libs` -> 0

PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config wayland-protocols --libs` -> 0

None of 'PKG_CONFIG_PATH' are defined in the environment, not changing global flags.
Run-time dependency wayland-protocols found: YES 1.20
Pkg-config binary for MachineChoice.HOST is cached.
Determining dependency 'xkbcommon' with pkg-config executable '/usr/bin/pkg-config'
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config --modversion xkbcommon` -> 0
1.0.2
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config --cflags xkbcommon` -> 0

PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config xkbcommon --libs` -> 0
-L/usr/lib -lxkbcommon
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config xkbcommon --libs` -> 0
-lxkbcommon
None of 'PKG_CONFIG_PATH' are defined in the environment, not changing global flags.
Run-time dependency xkbcommon found: YES 1.0.2
Pkg-config binary for MachineChoice.HOST is cached.
Determining dependency 'cairo' with pkg-config executable '/usr/bin/pkg-config'
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config --modversion cairo` -> 0
1.17.3
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config --cflags cairo` -> 0
-I/usr/include/cairo -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I/usr/include/pixman-1 -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/harfbuzz
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config cairo --libs` -> 0
-L/usr/lib -lcairo
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config cairo --libs` -> 0
-lcairo
None of 'PKG_CONFIG_PATH' are defined in the environment, not changing global flags.
Run-time dependency cairo found: YES 1.17.3
Pkg-config binary for MachineChoice.HOST is cached.
Determining dependency 'pango' with pkg-config executable '/usr/bin/pkg-config'
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config --modversion pango` -> 0
1.48.0
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config --cflags pango` -> 0
-I/usr/include/pango-1.0 -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I/usr/include/harfbuzz -I/usr/include/freetype2 -I/usr/include/libpng16 -pthread -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/fribidi -I/usr/include/cairo -I/usr/include/pixman-1
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config pango --libs` -> 0
-L/usr/lib -lpango-1.0 -lgobject-2.0 -lglib-2.0 -lharfbuzz
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config pango --libs` -> 0
-lpango-1.0 -lgobject-2.0 -lglib-2.0 -lharfbuzz
None of 'PKG_CONFIG_PATH' are defined in the environment, not changing global flags.
Run-time dependency pango found: YES 1.48.0
Pkg-config binary for MachineChoice.HOST is cached.
Determining dependency 'pangocairo' with pkg-config executable '/usr/bin/pkg-config'
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config --modversion pangocairo` -> 0
1.48.0
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config --cflags pangocairo` -> 0
-I/usr/include/pango-1.0 -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I/usr/include/harfbuzz -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/fribidi -I/usr/include/cairo -I/usr/include/pixman-1 -pthread
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config pangocairo --libs` -> 0
-L/usr/lib -lpangocairo-1.0 -lpango-1.0 -lgobject-2.0 -lglib-2.0 -lharfbuzz -lcairo
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config pangocairo --libs` -> 0
-lpangocairo-1.0 -lpango-1.0 -lgobject-2.0 -lglib-2.0 -lharfbuzz -lcairo
None of 'PKG_CONFIG_PATH' are defined in the environment, not changing global flags.
Run-time dependency pangocairo found: YES 1.48.0
Pkg-config binary for MachineChoice.HOST is cached.
Determining dependency 'gdk-pixbuf-2.0' with pkg-config executable '/usr/bin/pkg-config'
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config --modversion gdk-pixbuf-2.0` -> 0
2.42.0
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config --cflags gdk-pixbuf-2.0` -> 0
-I/usr/include/gdk-pixbuf-2.0 -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -pthread -I/usr/include/libmount -I/usr/include/blkid
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config gdk-pixbuf-2.0 --libs` -> 0
-L/usr/lib -lgdk_pixbuf-2.0 -lgobject-2.0 -lglib-2.0
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config gdk-pixbuf-2.0 --libs` -> 0
-lgdk_pixbuf-2.0 -lgobject-2.0 -lglib-2.0
None of 'PKG_CONFIG_PATH' are defined in the environment, not changing global flags.
Run-time dependency gdk-pixbuf-2.0 found: YES 2.42.0
Pkg-config binary for MachineChoice.HOST is cached.
Determining dependency 'pixman-1' with pkg-config executable '/usr/bin/pkg-config'
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config --modversion pixman-1` -> 0
0.40.0
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config --cflags pixman-1` -> 0
-I/usr/include/pixman-1
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config pixman-1 --libs` -> 0
-L/usr/lib -lpixman-1
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config pixman-1 --libs` -> 0
-lpixman-1
None of 'PKG_CONFIG_PATH' are defined in the environment, not changing global flags.
Run-time dependency pixman-1 found: YES 0.40.0
Pkg-config binary for MachineChoice.HOST is cached.
Determining dependency 'glesv2' with pkg-config executable '/usr/bin/pkg-config'
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config --modversion glesv2` -> 0
3.2
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config --cflags glesv2` -> 0

PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config glesv2 --libs` -> 0
-L/usr/lib -lGLESv2
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config glesv2 --libs` -> 0
-lGLESv2
None of 'PKG_CONFIG_PATH' are defined in the environment, not changing global flags.
Run-time dependency glesv2 found: YES 3.2
Pkg-config binary for MachineChoice.HOST is cached.
Determining dependency 'libevdev' with pkg-config executable '/usr/bin/pkg-config'
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config --modversion libevdev` -> 0
1.10.0
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config --cflags libevdev` -> 0
-I/usr/include/libevdev-1.0/
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config libevdev --libs` -> 0
-L/usr/lib -levdev
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config libevdev --libs` -> 0
-levdev
None of 'PKG_CONFIG_PATH' are defined in the environment, not changing global flags.
Run-time dependency libevdev found: YES 1.10.0
Pkg-config binary for MachineChoice.HOST is cached.
Determining dependency 'libinput' with pkg-config executable '/usr/bin/pkg-config'
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config --modversion libinput` -> 0
1.16.3
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config --cflags libinput` -> 0

PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config libinput --libs` -> 0
-L/usr/lib -linput
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config libinput --libs` -> 0
-linput
None of 'PKG_CONFIG_PATH' are defined in the environment, not changing global flags.
Run-time dependency libinput found: YES 1.16.3
Pkg-config binary for MachineChoice.HOST is cached.
Determining dependency 'libsystemd' with pkg-config executable '/usr/bin/pkg-config'
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config --modversion libsystemd` -> 0
246
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config --cflags libsystemd` -> 0

PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config libsystemd --libs` -> 0
-L/usr/lib -lsystemd
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config libsystemd --libs` -> 0
-lsystemd
None of 'PKG_CONFIG_PATH' are defined in the environment, not changing global flags.
Run-time dependency libsystemd found: YES 246
Pkg-config binary for MachineChoice.HOST is cached.
Determining dependency 'libelogind' with pkg-config executable '/usr/bin/pkg-config'
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config --modversion libelogind` -> 1

CMake binary for MachineChoice.BUILD is not cached
None of 'CMAKE' are defined in the environment, not changing global flags.
CMake binary missing from cross or native file, or env var undefined.
Trying a default CMake fallback at cmake
Found CMake: /usr/bin/cmake (3.19.0)
None of 'CMAKE_PREFIX_PATH' are defined in the environment, not changing global flags.
Extracting basic cmake information
Try CMake generator: auto
Calling CMake (['/usr/bin/cmake']) in /home/alexandre/workspace/archofficiel/community/sway/repos/community-x86_64/src/build/meson-private/cmake_libelogind with:
  - "--trace-expand"
  - "--trace-format=json-v1"
  - "--no-warn-unused-cli"
  - "--trace-redirect=cmake_trace.txt"
  - "-DCMAKE_TOOLCHAIN_FILE=/home/alexandre/workspace/archofficiel/community/sway/repos/community-x86_64/src/build/meson-private/cmake_libelogind/CMakeMesonToolchainFile.cmake"
  - "."
  -- Module search paths:    ['/', '/opt', '/usr', '/usr/local']
  -- CMake root:             /usr/share/cmake-3.19
  -- CMake architectures:    ['libpyldb-util.cpython-38-x86-64-linux-gnu.so', 'libpyldb-util.cpython-38-x86-64-linux-gnu.so.2', 'libpyldb-util.cpython-38-x86-64-linux-gnu.so.2.2.0', 'libpyside2.cpython-38-x86_64-linux-gnu.so', 'libpyside2.cpython-38-x86_64-linux-gnu.so.5.15', 'libpyside2.cpython-38-x86_64-linux-gnu.so.5.15.2', 'libpytalloc-util.cpython-38-x86-64-linux-gnu.so', 'libpytalloc-util.cpython-38-x86-64-linux-gnu.so.2', 'libpytalloc-util.cpython-38-x86-64-linux-gnu.so.2.3.1', 'libsamba-policy.cpython-38-x86-64-linux-gnu.so', 'libsamba-policy.cpython-38-x86-64-linux-gnu.so.0', 'libsamba-policy.cpython-38-x86-64-linux-gnu.so.0.0.1', 'libshiboken2.cpython-38-x86_64-linux-gnu.so', 'libshiboken2.cpython-38-x86_64-linux-gnu.so.5.15', 'libshiboken2.cpython-38-x86_64-linux-gnu.so.5.15.2']
  -- CMake lib search paths: ['lib', 'lib32', 'lib64', 'libx32', 'share', 'lib/libpyldb-util.cpython-38-x86-64-linux-gnu.so', 'lib/libpyldb-util.cpython-38-x86-64-linux-gnu.so.2', 'lib/libpyldb-util.cpython-38-x86-64-linux-gnu.so.2.2.0', 'lib/libpyside2.cpython-38-x86_64-linux-gnu.so', 'lib/libpyside2.cpython-38-x86_64-linux-gnu.so.5.15', 'lib/libpyside2.cpython-38-x86_64-linux-gnu.so.5.15.2', 'lib/libpytalloc-util.cpython-38-x86-64-linux-gnu.so', 'lib/libpytalloc-util.cpython-38-x86-64-linux-gnu.so.2', 'lib/libpytalloc-util.cpython-38-x86-64-linux-gnu.so.2.3.1', 'lib/libsamba-policy.cpython-38-x86-64-linux-gnu.so', 'lib/libsamba-policy.cpython-38-x86-64-linux-gnu.so.0', 'lib/libsamba-policy.cpython-38-x86-64-linux-gnu.so.0.0.1', 'lib/libshiboken2.cpython-38-x86_64-linux-gnu.so', 'lib/libshiboken2.cpython-38-x86_64-linux-gnu.so.5.15', 'lib/libshiboken2.cpython-38-x86_64-linux-gnu.so.5.15.2']
Preliminary CMake check failed. Aborting.
Run-time dependency libelogind found: NO (tried pkgconfig and cmake)
Pkg-config binary for MachineChoice.HOST is cached.
Determining dependency 'xcb' with pkg-config executable '/usr/bin/pkg-config'
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config --modversion xcb` -> 0
1.14
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config --cflags xcb` -> 0

PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config xcb --libs` -> 0
-L/usr/lib -lxcb
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config xcb --libs` -> 0
-lxcb
None of 'PKG_CONFIG_PATH' are defined in the environment, not changing global flags.
Run-time dependency xcb found: YES 1.14
Pkg-config binary for MachineChoice.HOST is cached.
Determining dependency 'bash-completion' with pkg-config executable '/usr/bin/pkg-config'
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config --modversion bash-completion` -> 1

CMake binary for MachineChoice.BUILD is cached.
None of 'CMAKE_PREFIX_PATH' are defined in the environment, not changing global flags.
Preliminary CMake check failed. Aborting.
Run-time dependency bash-completion found: NO (tried pkgconfig and cmake)
Pkg-config binary for MachineChoice.HOST is cached.
Determining dependency 'fish' with pkg-config executable '/usr/bin/pkg-config'
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config --modversion fish` -> 1

CMake binary for MachineChoice.BUILD is cached.
None of 'CMAKE_PREFIX_PATH' are defined in the environment, not changing global flags.
Preliminary CMake check failed. Aborting.
Run-time dependency fish found: NO (tried pkgconfig and cmake)
None of 'CC_LD' are defined in the environment, not changing global flags.
Running compile:
Working directory:  /home/alexandre/workspace/archofficiel/community/sway/repos/community-x86_64/src/build/meson-private/tmpu3onzikm
Command line:  ccache gcc /home/alexandre/workspace/archofficiel/community/sway/repos/community-x86_64/src/build/meson-private/tmpu3onzikm/testfile.c -o /home/alexandre/workspace/archofficiel/community/sway/repos/community-x86_64/src/build/meson-private/tmpu3onzikm/output.exe -pipe -march=x86-64 -mtune=generic -O2 -fno-plt -g -fvar-tracking-assignments -fdebug-prefix-map=/home/alexandre/workspace/archofficiel/community/sway/repos/community-x86_64/src=/usr/src/debug -D_FORTIFY_SOURCE=2 -D_FILE_OFFSET_BITS=64 -O0 -Wl,--start-group -lm -Wl,--end-group -Wl,--allow-shlib-undefined -Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now 

Code:
 int main(void) { return 0; }

Compiler stdout:
 
Compiler stderr:
 
Library m found: YES
None of 'CC_LD' are defined in the environment, not changing global flags.
Running compile:
Working directory:  /home/alexandre/workspace/archofficiel/community/sway/repos/community-x86_64/src/build/meson-private/tmp866nzq5n
Command line:  ccache gcc /home/alexandre/workspace/archofficiel/community/sway/repos/community-x86_64/src/build/meson-private/tmp866nzq5n/testfile.c -o /home/alexandre/workspace/archofficiel/community/sway/repos/community-x86_64/src/build/meson-private/tmp866nzq5n/output.exe -pipe -march=x86-64 -mtune=generic -O2 -fno-plt -g -fvar-tracking-assignments -fdebug-prefix-map=/home/alexandre/workspace/archofficiel/community/sway/repos/community-x86_64/src=/usr/src/debug -D_FORTIFY_SOURCE=2 -D_FILE_OFFSET_BITS=64 -O0 -Wl,--start-group -lrt -Wl,--end-group -Wl,--allow-shlib-undefined -Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now 

Code:
 int main(void) { return 0; }

Compiler stdout:
 
Compiler stderr:
 
Library rt found: YES
Neither a subproject directory nor a wlroots.wrap file was found.
Subproject  wlroots is buildable: NO (disabling)
Pkg-config binary for MachineChoice.HOST is cached.
Determining dependency 'wlroots' with pkg-config executable '/usr/bin/pkg-config'
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config --modversion wlroots` -> 0
0.12.0
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config --cflags wlroots` -> 0
-I/usr/include/libdrm -I/usr/include/pixman-1
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config wlroots --libs` -> 0
-L/usr/lib -lwlroots
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config wlroots --libs` -> 0
-lwlroots
None of 'PKG_CONFIG_PATH' are defined in the environment, not changing global flags.
Run-time dependency wlroots found: YES 0.12.0
Running compile:
Working directory:  /tmp/tmpnraerc25
Command line:  ccache gcc -I/usr/include/libdrm -I/usr/include/pixman-1 /tmp/tmpnraerc25/testfile.c -pipe -E -P -march=x86-64 -mtune=generic -O2 -fno-plt -g -fvar-tracking-assignments -fdebug-prefix-map=/home/alexandre/workspace/archofficiel/community/sway/repos/community-x86_64/src=/usr/src/debug -D_FORTIFY_SOURCE=2 -D_FILE_OFFSET_BITS=64 -P -O0 -std=c11 

Code:
 
        #include <wlr/config.h>
        #ifndef WLR_HAS_XWAYLAND
        # define WLR_HAS_XWAYLAND
        #endif
        "MESON_GET_DEFINE_DELIMITER"
WLR_HAS_XWAYLAND
Compiler stdout:
         "MESON_GET_DEFINE_DELIMITER"
1

Compiler stderr:
 
Fetching value of define "WLR_HAS_XWAYLAND" with dependency wlroots: 1 
Pkg-config binary for MachineChoice.BUILD is not cached.
None of 'PKG_CONFIG' are defined in the environment, not changing global flags.
Pkg-config binary missing from cross or native file, or env var undefined.
Trying a default Pkg-config fallback at pkg-config
Found pkg-config: /usr/bin/pkg-config (1.7.3)
Determining dependency 'scdoc' with pkg-config executable '/usr/bin/pkg-config'
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config --modversion scdoc` -> 0
1.11.0
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config --cflags scdoc` -> 0

PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config scdoc --libs` -> 0

PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config scdoc --libs` -> 0

None of 'PKG_CONFIG_PATH' are defined in the environment, not changing global flags.
Build-time dependency scdoc found: YES 1.11.0
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config --variable=scdoc scdoc` -> 0
/usr/bin/scdoc
Got pkgconfig variable scdoc : /usr/bin/scdoc
Program /usr/bin/scdoc found: YES (/usr/bin/scdoc)
Program sh found: YES (/usr/bin/sh)
Program git found: YES (/usr/bin/git)
Running command: /usr/bin/git rev-parse --short HEAD
--- stdout ---
7ea954aa367

--- stderr ---


Running command: /usr/bin/git rev-parse --abbrev-ref HEAD
--- stdout ---
master

--- stderr ---


Running compile:
Working directory:  /home/alexandre/workspace/archofficiel/community/sway/repos/community-x86_64/src/build/meson-private/tmp7bbkudqj
Command line:  ccache gcc /home/alexandre/workspace/archofficiel/community/sway/repos/community-x86_64/src/build/meson-private/tmp7bbkudqj/testfile.c -o /home/alexandre/workspace/archofficiel/community/sway/repos/community-x86_64/src/build/meson-private/tmp7bbkudqj/output.obj -pipe -c -march=x86-64 -mtune=generic -O2 -fno-plt -g -fvar-tracking-assignments -fdebug-prefix-map=/home/alexandre/workspace/archofficiel/community/sway/repos/community-x86_64/src=/usr/src/debug -D_FORTIFY_SOURCE=2 -D_FILE_OFFSET_BITS=64 -O0 -fmacro-prefix-map=/prefix/to/hide= 

Code:
 extern int i;
int i;

Compiler stdout:
 
Compiler stderr:
 
Compiler for C supports arguments -fmacro-prefix-map=/prefix/to/hide=: YES 
Configuring config.h using configuration
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config --variable=pkgdatadir wayland-protocols` -> 0
//usr/share/wayland-protocols
Got pkgconfig variable pkgdatadir : //usr/share/wayland-protocols
Pkg-config binary for MachineChoice.BUILD is cached.
Determining dependency 'wayland-scanner' with pkg-config executable '/usr/bin/pkg-config'
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config --modversion wayland-scanner` -> 0
1.18.0
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config --cflags wayland-scanner` -> 0

PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config wayland-scanner --libs` -> 0

PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config wayland-scanner --libs` -> 0

None of 'PKG_CONFIG_PATH' are defined in the environment, not changing global flags.
Build-time dependency wayland-scanner found: YES 1.18.0
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config --variable=wayland_scanner wayland-scanner` -> 0
/usr/bin/wayland-scanner
Got pkgconfig variable wayland_scanner : /usr/bin/wayland-scanner
Program /usr/bin/wayland-scanner found: YES (/usr/bin/wayland-scanner)
Configuring config using configuration
Build targets in project: 38

sway 1.5.1

      xwayland: YES
    gdk-pixbuf: YES
       systemd: YES
       elogind: NO
          tray: YES
     man-pages: YES

  Subprojects
       wlroots: NO Neither a subproject directory nor a wlroots.wrap file was found.

Option werror is: false [default: true]
Found ninja-1.10.1 at /usr/bin/ninja
None of 'STRIP' are defined in the environment, not changing global flags.
None of 'STRIP' are defined in the environment, not changing global flags.
```

and full `install-logs.txt`:

```
# List of files installed by Meson
# Does not contain files installed by custom scripts.
/home/alexandre/workspace/archofficiel/community/sway/repos/community-x86_64/pkg/sway/usr/share/man/man1/sway.1
/home/alexandre/workspace/archofficiel/community/sway/repos/community-x86_64/pkg/sway/usr/share/man/man1
/home/alexandre/workspace/archofficiel/community/sway/repos/community-x86_64/pkg/sway/usr/share/man
/home/alexandre/workspace/archofficiel/community/sway/repos/community-x86_64/pkg/sway/usr/share
/home/alexandre/workspace/archofficiel/community/sway/repos/community-x86_64/pkg/sway/usr
```

--- a/mesonbuild/scripts/depfixer.py
+++ b/mesonbuild/scripts/depfixer.py
@@ -836,7 +836,7 @@ def fix_rpath(fname: str, rpath_dirs_to_remove: T.Set[bytes], new_rpath: T.Union
     # on non-mac platforms. That can be expensive on some Windows machines
     # (up to 30ms), which is significant with --only-changed. For details, see:
     # https://github.com/mesonbuild/meson/pull/6612#discussion_r378581401
-    if INSTALL_NAME_TOOL is False:
+    if system == 'darwin' and INSTALL_NAME_TOOL is False:
         INSTALL_NAME_TOOL = bool(shutil.which('install_name_tool'))
     if INSTALL_NAME_TOOL:
         if isinstance(new_rpath, bytes):

--- a/mesonbuild/scripts/depfixer.py
+++ b/mesonbuild/scripts/depfixer.py
@@ -836,9 +836,10 @@ def fix_rpath(fname: str, rpath_dirs_to_remove: T.Set[bytes], new_rpath: T.Union
     # on non-mac platforms. That can be expensive on some Windows machines
     # (up to 30ms), which is significant with --only-changed. For details, see:
     # https://github.com/mesonbuild/meson/pull/6612#discussion_r378581401
-    if INSTALL_NAME_TOOL is False:
-        INSTALL_NAME_TOOL = bool(shutil.which('install_name_tool'))
-    if INSTALL_NAME_TOOL:
-        if isinstance(new_rpath, bytes):
-            new_rpath = new_rpath.decode('utf8')
-        fix_darwin(fname, rpath_dirs_to_remove, new_rpath, final_path, install_name_mappings)
+    if system == 'darwin':
+        if INSTALL_NAME_TOOL is False:
+            INSTALL_NAME_TOOL = bool(shutil.which('install_name_tool'))
+        if INSTALL_NAME_TOOL:
+            if isinstance(new_rpath, bytes):
+                new_rpath = new_rpath.decode('utf8')
+            fix_darwin(fname, rpath_dirs_to_remove, new_rpath, final_path, install_name_mappings)

--- a/unittests/internaltests.py
+++ b/unittests/internaltests.py
@@ -25,6 +25,7 @@ import mesonbuild.dependencies.factory
 import mesonbuild.envconfig
 import mesonbuild.environment
 import mesonbuild.modules.gnome
+import mesonbuild.scripts.depfixer
 import mesonbuild.scripts.env2mfile
 from mesonbuild import coredata
 from mesonbuild.compilers.c import ClangCCompiler, GnuCCompiler
@@ -2233,3 +2234,13 @@ class InternalTests(unittest.TestCase):
                 self.assertEqual(actual.compile_args, expected.compile_args)
                 self.assertEqual(actual.link_args, expected.link_args)
                 self.assertEqual(actual.cmake, expected.cmake)
+
+    def test_depfixer_skips_install_name_tool_on_non_darwin(self) -> None:
+        with tempfile.NamedTemporaryFile(suffix='.so') as f:
+            f.write(b'not an elf file')
+            f.flush()
+            with mock.patch('mesonbuild.scripts.depfixer.INSTALL_NAME_TOOL', True), \
+                 mock.patch('mesonbuild.scripts.depfixer.fix_darwin') as mock_fix_darwin:
+                mesonbuild.scripts.depfixer.fix_rpath(
+                    f.name, set(), '', '', {}, system='linux', verbose=False)
+                mock_fix_darwin.assert_not_called()

--- a/unittests/internaltests.py
+++ b/unittests/internaltests.py
@@ -26,6 +26,7 @@ import mesonbuild.envconfig
 import mesonbuild.environment
 import mesonbuild.modules.cuda
 import mesonbuild.modules.gnome
+import mesonbuild.scripts.depfixer
 import mesonbuild.scripts.env2mfile
 from mesonbuild import coredata
 from mesonbuild.compilers.c import ClangCCompiler, GnuCCompiler
@@ -2407,3 +2408,14 @@ Thread model: posix'''), '21.9.0')
             with self.subTest(cuda_version=cuda_version, arch_list=arch_list):
                 with self.assertRaisesRegex(InvalidArguments, message):
                     flags(cuda_version, arch_list)
+
+    def test_depfixer_skips_install_name_tool_on_non_darwin(self) -> None:
+        with tempfile.TemporaryDirectory() as tmpdir:
+            fname = os.path.join(tmpdir, 'libfoo.so')
+            with open(fname, 'wb') as f:
+                f.write(b'not an elf file')
+            with mock.patch('mesonbuild.scripts.depfixer.INSTALL_NAME_TOOL', True), \
+                 mock.patch('mesonbuild.scripts.depfixer.fix_darwin') as mock_fix_darwin:
+                mesonbuild.scripts.depfixer.fix_rpath(
+                    fname, set(), '', '', {}, system='linux', verbose=False)
+                mock_fix_darwin.assert_not_called()

